docs/whitepaper/whitepaper.md

```diff
@@ -0,0 +1,591 @@
+        - [4.1.1.1.3 Configuration Schema - API Server](#41113-configuration-schema---api-server)
+        - [4.1.1.1.4 Plugin Loading/Validation](#41114-plugin-loadingvalidation)
+      - [4.1.1.2 cmd-api-server](#4112-cmd-api-server)
+      - [4.1.1.3 cmd-validator](#4113-cmd-validator)
+      - [4.1.1.4 sdk-javascript](#4114-sdk-javascript)
+      - [4.1.1.5 keychain](#4115-keychain)
+      - [4.1.1.7 tracing](#4117-tracing)
+      - [4.1.1.8 audit](#4118-audit)
+      - [4.1.1.9 document-storage](#4119-document-storage)
+      - [4.1.1.10 relational-storage](#41110-relational-storage)
+      - [4.1.1.11 immutable-storage](#41111-immutable-storage)
+    - [4.1.2 Deployment Diagram](#412-deployment-diagram)
+    - [4.1.3 Component Diagram](#413-component-diagram)
+    - [4.1.4 Class Diagram](#414-class-diagram)
+    - [4.1.5 Sequence Diagram - Transactions](#415-sequence-diagram---transactions)
+  - [4.2 Transaction Protocol Specification](#42-transaction-protocol-specification)
+    - [4.2.1 Handshake Mechanism](#421-handshake-mechanism)
+    - [4.1.9 Transaction Protocol Negotiation](#419-transaction-protocol-negotiation)
+  - [4.3 Plugin Architecture](#43-plugin-architecture)
+    - [4.3.1 Ledger Connector Plugins](#431-ledger-connector-plugins)
+    - [4.3.2 Identity Federation Plugins](#432-identity-federation-plugins)
+      - [4.3.1.1 X.509 Certificate Plugin](#4311-x509-certificate-plugin)
+    - [4.3.3 Storage Plugins](#433-storage-plugins)
+- [5. Identities, Authentication, Authorization](#5-identities-authentication-authorization)
+  - [5.1 Transaction Signing Modes, Key Ownership](#51-transaction-signing-modes-key-ownership)
+    - [5.1.1 Client-side Transaction Signing](#511-client-side-transaction-signing)
+    - [5.1.2 Server-side Transaction Signing](#512-server-side-transaction-signing)
+  - [5.2 Open ID Connect Provider, Identity Provider](#52-open-id-connect-provider-identity-provider)
+  - [5.3 Server-side Keychain for Web Applications](#53-server-side-keychain-for-web-applications)
+- [6. Terminology](#6-terminology)
+- [7. References](#7-references)
+
+
+# 1. Abstract
+
+Blockchain technologies are growing in usage, but fragmentation is a big problem that may hinder reaching critical levels of adoption in the future.
+
+We propose a protocol and it's implementation to connect as many of them as possible in an attempt to solve the fragmentation problem by creating a heterogeneous system architecture <sup>[1](#7-references)</sup> similar to what the banking system has where each institution can have their own internal implementation of financial transactions and yet cross-institution transfers are a common and well supported use case.
+
+We do not limit ourselves to financial transactions, but instead aim to remain unopinionated and flexible for future extensions to be a catalyst of innovation rather than slowing it down.
+
+# 2. Example Use Cases
```

Comment:
Fujimoto-san and Takeuchi-san:  could you all PR this section with some of your suggestions?

Reply:
Yes, please feel free to propose and/or make changes/add/remove things etc. 
The document is still very much just a draft and I am more than happy to have even entire sections removed if not everyone thinks they make sense (which is very much a possibility I might add). :-)